enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,597 @@
+2. The user defines the `installConfig.controlPlane` field with `2` replicas.
+3. The user then enters the new field `installConfig.arbiterNode` defines the
+   arbiter node.
+4. The user then enters the machine information for
+   `installConfig.controlPlane.platform.baremetal` and
+   `installConfig.arbiterNode.platform.baremetal`
+5. The installer creates a new `arbiter` MachineSet with a replica of `1` and
+   checks the control plane replicas is `2` or higher
+6. The installer sets the control plane and infrastructure topology to
+   `HighlyAvailableArbiter`
+7. With the object `arbiterNode` in the install config, the installer creates
+   the machine object with the correct node labels and machine config resources
+   for ignition bootstrap files.
+8. The installer applies the taint to the arbiter MachineSet
+9. The user can make any alterations to the node machine type to use less
+   powerful machines.
+10. The user then begins the install via `openshift-install create cluster`
+
+#### During Install
+
+1. The CEO will watch for new masters and the arbiter role
+2. CEO will create the operand for the etcd deployments that have tolerations
+   for the arbiter
+3. Operators that have tolerations for the arbiter should be scheduled on the
+   node
+4. The install should proceed as normal
+
+### API Extensions
+
+#### Installer API Change
+
+The `installConfig` will include a `installConfig.arbiterNode` object for a
+`MachinePool` to configure arbiter infra structure.
+
+#### OCP Config API Change
+
+The Infrastructure config fields for `ControlPlaneTopology` and
+`InfrastructureTopology` will support the new value of `HighlyAvailableArbiter`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+At the time being there is no impact on Hypershift since this edge deployment
+will require running the control plane.
+
+#### Standalone Clusters
+
+This change is relevant to standalone deployments of OpenShift at the edge or
+datacenters. This enhancement specifically deals with this type of deployment.
+
+#### Single-node Deployments or MicroShift
+
+This change does not effect Single Node or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+Currently there are some behavior unknowns, we will need to test out our POC to
+validate some of the desires in this proposal. In it's current version this
+proposal is not exhaustive but will be filled out as we implement these goals.
+
+We currently expect this feature to mainly be used by `baremetal` installs, or
+specialized hardware that is built to take advantage of this type of
+configuration. In the current design we make two paths for cloud and baremetal
+installs in the installer. However, the cloud install is primarily for testing,
+this might mean that we simplify the installer changes if we are the only ones
+using cloud installs, since we can simply alter the manifests in the pipeline
+with out needing to change the installer.
```

Comment:
I would seriously question whether the work saved in manually setting up tests for etcd justifies the extra work in supporting this on cloud platforms.
Not only are cloud platforms an extra thing to support, but installing the arbiter node via a MachineSet in a platform-independent way complicates the baremetal implementation, and the Assisted/ABI implementations as well.

Reply:
I understand the concern, I'll try to capture this better in the documentation, but there is value for cloud deployment, especially as a TechPreview iteration of this feature. Currently there are some unknowns around where users are with the hardware they will be using, so having a cloud environment would be easier for them to deploy and spin the tires on it as well as make testing and validation a bit easier for us.

I went ahead and updated the doc to reflect the fact that the arbiter node is being installed in the same way as the master nodes, would that help smooth out your concerns?

Reply:
We realigned the focus to just baremetal for the time being, the doc has been updated to focus only on baremetal.